src/OSInfo.cpp

```diff
@@ -225,24 +225,52 @@ QString OSInfo::getOSVersion() {
 	           buildno ? buildno : "unknown");
 #else
 #ifdef Q_OS_LINUX
-	QProcess qp;
-	QStringList args;
-	args << QLatin1String("-s");
-	args << QLatin1String("-d");
-	qp.start(QLatin1String("lsb_release"), args);
-	if (qp.waitForFinished(5000)) {
-		os = QString::fromUtf8(qp.readAll()).simplified();
-		if (os.startsWith(QLatin1Char('"')) && os.endsWith(QLatin1Char('"')))
-			os = os.mid(1, os.length() - 2).trimmed();
+	QFile qf(QLatin1String("/etc/os-release"));
+	if (!qf.open(QIODevice::ReadOnly)) {
+		qf.setFileName(QLatin1String("/usr/lib/os-release"));
+		qf.open(QIODevice::ReadOnly);
 	}
-	if (os.isEmpty())
-		qWarning("OSInfo: Failed to execute lsb_release");
+	if (qf.isReadable()) {
+		qDebug() << "OSInfo: Retrieving version from" << qf.fileName();
+		bool foundPreferred = false;
+		QTextStream ts(&qf);
+		while (!ts.atEnd()) {
+			QString line = ts.readLine();
+			if (line.startsWith(QLatin1String("PRETTY_NAME"))) {
+				foundPreferred = true;
+			}
+
+			if (foundPreferred || line.startsWith(QLatin1String("NAME"))) {
+				os = line.mid(line.indexOf(QLatin1Char('=')) + 1);
+			}
 
-	qp.terminate();
-	if (! qp.waitForFinished(1000))
-		qp.kill();
+			if (foundPreferred) {
+				break;
+			}
+		}
+
+		qf.close();
+	}
+
+	if (os.isEmpty()) {
```

Comment:
Can we put this code block into a method? We are using methods way too scarcely. They name code blocks, help us structure code and improve readability. We have way too many too long methods.

Another block is the os-release check.

Reply:
I agree.